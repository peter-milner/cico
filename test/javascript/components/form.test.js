import React from 'react'
import { act } from 'react-dom/test-utils'
import { mount } from 'enzyme'
import * as axios from 'axios'

jest.mock('axios')
axios.post.mockImplementation(() => Promise.resolve({ status: 200, data: [] }))

import Form from '../../../app/javascript/components/form'

const defaultProps = {
    setNotificationMessage: jest.fn(),
    updateEvents: jest.fn()
}

const render = (props) => {
    const newProps = Object.assign({}, defaultProps, props)
    return mount(
        <Form {...newProps} />
    )
}

describe('<Form />', () => {
    test('snapshot', () => {
        const rendered = render()
        expect(rendered).toMatchSnapshot()
    })

    test('axios sends request with correct form data', () => {
        const rendered = render()
        const name = 'test123'
        const spy = jest.spyOn(axios, 'post')

        rendered.find('input').simulate('change', { target: { value: name } })
        rendered.find('button').first().simulate('click')
        rendered.find('form').simulate('submit')
        
        expect(spy).toHaveBeenCalledWith('/events', {'name': name, 'status': 1})
    })
})