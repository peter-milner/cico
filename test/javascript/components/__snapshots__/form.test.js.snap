// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> snapshot 1`] = `ReactWrapper {}`;
